### PR TITLE
Csv seeds

### PR DESCRIPTION
**Pull request by efuchsman**
Database Setup
model class setup
model spec setup
creation relationships and validations for models
create status enums for invoices, invoice_items and transactions
move shouldamatchers to rails_helper
add simplecov.start to rails_helper
testing at 100%....so far...

### CODE REVIEW

**Review by DarbySmith — APPROVED** [review]
Looks good! One thing that we could possibly refactor later is adding different validations for numericality and such.

File: app/models/transaction.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Transaction < ApplicationRecord
+  belongs_to :invoice
+
+  validates_presence_of :credit_card_number, :result, :invoice_id
+
+  enum result: {success: 0, failed: 1}
```
Comment (bkeener7) [inline app/models/transaction.rb:6]: Do we need to test these `enum` lines?

**Review by bkeener7 — APPROVED** [review]
Left one comment, otherwise this feels like a great kickoff to the project.

File: app/models/transaction.rb (hunk 1 above)
Comment (DarbySmith) [reply app/models/transaction.rb:6]: That is a good question. I found this [stack overflow](https://stackoverflow.com/questions/29550958/how-should-i-test-rails-enums), so I think we may have to test this later.

**Pull request merged**